coredb-operator/tests/integration_tests.rs

```diff
@@ -0,0 +1,119 @@
+        let timeout_seconds = 30;
+
+        // Apply a basic configuration of CoreDB
+        println!("Creating CoreDB resource {}", name);
+        let coredbs: Api<CoreDB> = Api::namespaced(client.clone(), namespace);
+        let coredb_json = serde_json::json!({
+            "apiVersion": API_VERSION,
+            "kind": kind,
+            "metadata": {
+                "name": name
+            },
+            "spec": {
+                "replicas": replicas
+            }
+        });
+        let params = PatchParams::apply("coredb-integration-test");
+        let patch = Patch::Apply(&coredb_json);
+        let _coredb_resource = coredbs.patch(name, &params, &patch).await;
+
+        // Wait for Pod to be created
+
+        let pod_name = format!("{}-0", name);
+        println!("Waiting for pod to be running: {}", pod_name);
+        let pods: Api<Pod> = Api::namespaced(client.clone(), namespace);
+        if let Err(_) = tokio::time::timeout(
+            std::time::Duration::from_secs(timeout_seconds),
+            await_condition(pods, &pod_name, conditions::is_pod_running()),
+        )
+        .await
+        {
+            panic!(
+                "\n\nERROR: Did not find the pod {} to be running after waiting for {} seconds\n\n",
+                pod_name, timeout_seconds
+            )
+        }
+    }
+
+    async fn kube_client() -> kube::Client {
+        // Initialize the Kubernetes client
+        let client_future = Client::try_default();
+        let client = match client_future.await {
+            Ok(wrapped_client) => wrapped_client,
+            Err(_error) => panic!("Please configure your Kubernetes Context"),
+        };
+        // Get the name of the currently selected namespace
+        let selected_namespace = Config::infer().await.unwrap().default_namespace;
+
+        // Next, check that the currently selected namespace is labeled
+        // to allow the running of tests.
+
+        // List the namespaces with the specified labels
+        let namespaces: Api<Namespace> = Api::all(client.clone());
+        let namespace = namespaces.get(&selected_namespace).await.unwrap();
+        let labels = namespace.metadata.labels.unwrap();
+        assert!(
+            labels.contains_key("safe-to-run-coredb-tests"),
+            "expected to find label 'safe-to-run-coredb-tests'"
+        );
+        assert_eq!(
+            labels["safe-to-run-coredb-tests"], "true",
+            "expected to find label 'safe-to-run-coredb-tests' with value 'true'"
+        );
+
+        // Check that the CRD is installed
+        let crds: Api<CustomResourceDefinition> = Api::all(client.clone());
+        if let Err(_) = tokio::time::timeout(
+            std::time::Duration::from_secs(2),
+            await_condition(crds, "coredbs.kube.rs", conditions::is_crd_established()),
+        )
+        .await
+        {
+            panic!("\n\nERROR: Did not find the CRD to be installed.\n\n")
+        }
```

Comment:
For example same pattern here to wait on CRD. kube-rs also offers custom conditions, so can write a function and wait until that is satisfied.

https://docs.rs/kube/latest/kube/runtime/wait/trait.Condition.html